### PR TITLE
Handle linters when building (CRAFT-325).

### PR DESCRIPTION
**Pull request by facundobatista**
Show errors and warnings. Fail if there are some errors, unless `--force` was indicated to `pack` (not to `build`, which we want to not touch until we remove it).

Note this depends on #449.

### CODE REVIEW

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -146,12 +146,53 @@ def __init__(self, args, config):
...
+                logger.info(template, result.name, result.text, result.url)
+        if linters.ERRORS in lint_results_by_outcome:
+            logger.info("Lint Errors:")
+            for result in lint_results_by_outcome[linters.ERRORS]:
+                logger.info(template, result.name, result.text, result.url)
+            if self.force_packing:
+                logger.info("Packing anyway as requested.")
+            else:
+                raise CommandError(
+                    "Exiting after lint errors (use --force to pack anyway).", retcode=2
```
Comment (sergiusens) [inline charmcraft/commands/build.py:193]:
```suggestion
                    "Exiting after lint errors (use --force to override).", retcode=2
```

**Review by sergiusens — APPROVED** [review]

File: charmcraft/commands/build.py (hunk 1 above)
Comment (cjp256) [reply charmcraft/commands/build.py:193]: Maybe `Aborting due to lint errors ...` ?  Exiting is a bit ambiguous.

**Review by cjp256 — APPROVED** [review]
LGTM !

File: charmcraft/commands/build.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/build.py:193]: Like it!

**Pull request merged**